### PR TITLE
light wallet integration

### PR DESCRIPTION
**Pull request by 7flash**

### CODE REVIEW

**Comment by 7flash** [comment]
#175

File: src/boot/api.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ const signTransaction = async function (actions) {
         expireSeconds: 30
       })
       transactionId = result.transaction_id
+    } else if (typeof window.LightWalletChannel === 'object') {
+      const firstAction = actions[0]
+      transactionId = await this.$lightWallet.sendTransaction(firstAction)
```
Comment (gregory-latinier) [inline src/boot/api.js:34]: You can't only sign the first action otherwise you won't be able to fully use the app.
For example you can claim multiple periods, becoming a member requires multiple actions

File: src/layouts/MainLayout.vue [hunk 2]
```diff
@@ -64,6 +64,9 @@ export default {
 
     this.initNotifications()
     await this.fetchPeriods()
+
+    this.lightWalletLogin()
```
Comment (gregory-latinier) [inline src/layouts/MainLayout.vue:68]: please move this into the autoLogin method
it takes care of selecting the wallet for the user

here you will always login with the lightwallet and then use the other wallets
I think that in this situation we can have 2 login light wallet and Anchor for instance

Comment (7flash) [reply src/layouts/MainLayout.vue:68]: fixed

File: src/boot/light-wallet.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-new-func */
+class LightWallet {
+  resolveChannelResponse = Function()
+
+  waitChannelResponse () {
+    return new Promise(
+      function (resolve) {
+        this.resolveChannelResponse = response => {
+          console.log('channel response: ' + response)
```
[the remark this answers is not in the document]
Comment (7flash) [reply src/boot/light-wallet.js]: fixed

File: src/boot/api.js (hunk 1 above)
Comment (7flash) [reply src/boot/api.js]: fixed

Comment (7flash) [reply src/boot/api.js]: https://github.com/JoinSEEDS/seeds_light_wallet/pull/341/commits/c3a9814ee3a2ce795d607f6012b63847ef2f0955

**Comment by 7flash** [comment]
Peek 2020-11-17 14-31

Still has problem - shows account name only after going to another page and back
Can you help with that @gregory-latinier please?

File: src/layouts/MainLayout.vue [hunk 4]
```diff
@@ -26,7 +26,7 @@ export default {
     ...mapMutations('notifications', ['initNotifications', 'unmarkRead', 'unmarkNew']),
     ...mapMutations('layout', ['setShowRightSidebar', 'setRightSidebarType']),
     ...mapActions('periods', ['fetchPeriods']),
-    ...mapActions('accounts', ['autoLogin']),
+    ...mapActions('accounts', ['autoLogin', 'lightWalletLogin']),
```
Comment (gregory-latinier) [inline src/layouts/MainLayout.vue:29]: Why do you add this method if you're not using it ?

File: src/layouts/MainLayout.vue [hunk 5]
```diff
@@ -64,6 +64,7 @@ export default {
 
     this.initNotifications()
     await this.fetchPeriods()
+
```
Comment (gregory-latinier) [inline src/layouts/MainLayout.vue:67]: Unnecessary change

File: src/store/accounts/actions.js [hunk 6]
```diff
@@ -22,20 +41,32 @@ export const loginWallet = async function ({ commit, dispatch }, { idx, returnUr
     }
     localStorage.setItem('known-user', true)
     if (this.$router.currentRoute.path !== (returnUrl || '/dashboard')) {
-      await this.$router.push({ path: (returnUrl || '/dashboard') })
+      await this.$router.push({ path: returnUrl || '/dashboard' })
```
Comment (gregory-latinier) [inline src/store/accounts/actions.js:44]: Please remove your formatting
Only change the code that is necessary

**Review by gregory-latinier — CHANGES_REQUESTED** [review]

**Comment by 7flash** [comment]
@gregory-latinier Please merge

**Comment by 7flash** [comment]
Cool thanks :+1: :clinking_glasses:
